platform/database/factories/CreditTransactionFactory.php

```
@@ -0,0 +1,70 @@
+    {
+        return $this->state(fn (array $attributes) => [
+            'type' => CreditTransactionType::InitialGrant,
+            'amount' => 5,
+            'balance_after' => 5,
+            'description' => 'Welcome credits',
+        ]);
+    }
+
+    public function purchase(int $amount = 100): static
+    {
+        return $this->state(fn (array $attributes) => [
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attributes'. (undefined)

(UnusedFormalParameter)